### PR TITLE
Platform - UI: A11y Updates

### PR DESCRIPTION
**Pull request by deetz99**
*Description of changes:*

- fix aria labels
- add aria-required, aria-invalid, aria-describedby and aria-disabled to relevant input fields
- update input types eg: text -> email/url/tel (need to muck with zod to change string to number for fax/extension) 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the strr license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-238-fwn6zr50.web.app

**Review by thorwolpert — APPROVED** [review]

File: strr-platform-web/app/components/connect/form/address/Index.vue [hunk 1]
```diff
@@ -40,100 +37,146 @@ const addressComplete = () => {
...
           size="lg"
           :color="country ? 'primary' : 'gray'"
           :placeholder="$t('label.country')"
+          :aria-label="$t('label.country')"
           :options="countries"
           :ui-menu="{ label: country ? 'text-gray-900' : !!error? 'text-red-600': 'text-gray-700'}"
           option-attribute="name"
           data-testid="address-country"
+          :aria-required="true"
+          :aria-invalid="error !== undefined"
```
Comment (kialj876) [inline strr-platform-web/app/components/connect/form/address/Index.vue:60]: nice

File: strr-platform-web/app/components/connect/form/address/Index.vue [hunk 2]
```diff
@@ -40,100 +37,146 @@ const addressComplete = () => {
...
+          :aria-describedby="schemaPrefix + 'country-' + addId"
         />
-      </UFormGroup>
-    </div>
-    <div class="flex flex-row">
-      <UFormGroup :name="schemaPrefix + 'street'" :hint="t('text.streetHint')" class="grow">
+      </template>
+
+      <template #error="{ error }">
+        <span :id="schemaPrefix + 'country-' + addId">
```
Comment (kialj876) [inline strr-platform-web/app/components/connect/form/address/Index.vue:66]: just curious why the id is needed here and below? Is this for an AA thing?

**Review by kialj876 — APPROVED** [review]
Nice work, lots of good AA cleanup in here

File: strr-platform-web/app/components/connect/form/address/Index.vue (hunk 2 above)
Comment (deetz99) [reply strr-platform-web/app/components/connect/form/address/Index.vue:66]: Similar to the label element where you tie a label to an input by an id, you can specify additional details with the [aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) attribute. So adding this id and matching it to the aria-describedby on the input tells the screen reader user there is additional info for whatever input they're on.

**Pull request merged**